File: frontend-admin/src/types/attendance.ts
export type AttendanceStatus = 'ATTENDING' | 'ABSENT' | 'LATE' | 'NO_RESPONSE'

export type AttendanceRow = {
  id: string
  event_id: string
  child_id: string
  status: AttendanceStatus
  note: string | null
  responded_at: string | null
  child_first_name: string
  child_last_name: string
}

export type EventAttendance = {
  event_id: string
  summary: { attending: number; absent: number; late: number; no_response: number; total: number }
  attendances: AttendanceRow[]
}

File: backend/app/schemas/attendance.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

AttendanceStatus = Literal["ATTENDING", "ABSENT", "LATE", "NO_RESPONSE"]


class AttendanceUpsert(BaseModel):
    event_id: uuid.UUID
    child_id: uuid.UUID
    status: AttendanceStatus
    note: str | None = Field(default=None, max_length=500)


class AttendanceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    event_id: uuid.UUID
    child_id: uuid.UUID
    status: AttendanceStatus
    note: str | None
    responded_at: datetime | None


class AttendanceRowResponse(AttendanceResponse):
    child_first_name: str
    child_last_name: str


class AttendanceSummary(BaseModel):
    attending: int
    absent: int
    late: int
    no_response: int
    total: int


class EventAttendanceResponse(BaseModel):
    event_id: uuid.UUID
    summary: AttendanceSummary
    attendances: list[AttendanceRowResponse]
